### ISSUE TITLE
MSAL 1.4.0 was released, and is causing infinite redirection loop

### ISSUE THREAD

**Issue by kumarunster** [AzureAD/microsoft-authentication-library-for-js#2185]
## Library
- [x] `msal@1.4.0` 
- [ ] `@azure/msal-browser@2.x.x`
- [ ] `@azure/msal-angular@0.x.x`
- [ ] `@azure/msal-angular@1.x.x`
- [ ] `@azure/msal-angularjs@1.x.x`

## Framework
React

## Description
We noticed an upgrade of msal to the version 1.4.0 today in our dev environment. It was released at the same day (25. Aug. 2020). The login with underlying msal library does not work anymore and is ending in infinite redirection loop to login.microsoft.com. 

More, after short search on this project side I cannot find any migration guide or release note to the released version. Was this version released by accident? Could someone clarify?

For now we changed the version back to the latest 1.3.x and everything works fine.

## Security
- [ ] Is this issue security related?

## Regression
- [ ] Did this behavior work before? 
Version:

## MSAL Configuration

```js
// Provide configuration values here.
// For Azure B2C issues, please include your policies.
```

## Reproduction steps
```js
// Provide relevant code snippets here.
// For Azure B2C issues, please include your policies.
```

## Browsers/Environment
- [x] Chrome
- [ ] Firefox
- [ ] Edge
- [ ] Safari
- [ ] IE
- [ ] Other (Please add browser name here)

**Comment by technical-boy** [comment]
Hi @kumarunster . Can you please provide the following information:

1. Your MSAL configuration
2. Code snippets demonstrating your usage (specifically interested in usage of acquireTokenSilent and loginRedirect/acquireTokenRedirect)
3. Are there any errors from the browser console?
4. Logs from the MSAL Logger
5. Have you performed any debugging steps to determine why MSAL is looping?
6. Does this happen in all browsers, or just Chrome?
7. Are third-party cookies enabled in your browser?

The changelog can be found [here](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-core/changelog.md).

**Comment by kumarunster** [comment]
Hi @technical-boy, thanks for your reply. We are using msal with [react-aad-msal](https://github.com/syncweek-react-aad/react-aad#readme).

hereby our provider instantiation:
```typescript
            authProvider = new MsalAuthProvider(
            {
              auth: {
                authority: authority,
                clientId: clientId,
                postLogoutRedirectUri: window.location.origin,
                redirectUri: redirectUri,
                validateAuthority: false,
                
                // After being redirected to the "redirectUri" page, should user
                // be redirected back to the Url where their login originated from?
                navigateToLoginRequestUrl: false
              },
              cache: {
                cacheLocation: 'localStorage',
                storeAuthStateInCookie: false,
              },
              system: {
                tokenRenewalOffsetSeconds: 600
              }
            },
            {
              scopes: scopes,
            },
            {
            loginType
            }
          );

```

the scopes we are using are 
in AAD case: `https://graph.microsoft.com/openid,https://graph.microsoft.com/profile`
in B2C case: `https://xyz.onmicrosoft.com/api/openid,https://xyz.onmicrosoft.com/api/profile`

so far we did't any further debugging why exactly the loop is caused, we quickly reverted the version and everything is working again as expected.

**Comment by technical-boy** [comment]
Hi, @kumarunster, thanks for following up. Could you please use the MSAL Logger, reproduce the bug and send us the logs? Thanks.

**Comment by kumarunster** [comment]
here is the log file

[localhost-1598434880293.log](https://github.com/AzureAD/microsoft-authentication-library-for-js/files/5129006/localhost-1598434880293.log)

seems something to be wrong with cached id token or/and with access to cached token by key. MSAL returns id token with access token as **tokenType** instead, although a correct id token exists in local storage multiple times. Hence this is the cause for infinite login loop. Could it be related to the changes in [PR#2022](https://github.com/AzureAD/microsoft-authentication-library-for-js/pull/2022)? Could another method be used to determine the correctness of tokenType?

AuthCache.ts:
(see screenshot from local storage. Key is without scopes).
<img width="956" alt="Bildschirmfoto 2020-08-26 um 11 56 13" src="https://user-images.githubusercontent.com/2104820/91290771-570f0680-e794-11ea-940a-0964893c30ee.png">


UserAgentApplication.ts#getCachedToken:
<img width="1398" alt="Bildschirmfoto 2020-08-26 um 11 59 12" src="https://user-images.githubusercontent.com/2104820/91290818-68f0a980-e794-11ea-998f-c962e72e6a93.png">

LocalStorage:
<img width="1134" alt="Bildschirmfoto 2020-08-26 um 12 02 00" src="https://user-images.githubusercontent.com/2104820/91291086-d4d31200-e794-11ea-9400-5b851e73a4bc.png">

**Comment by technical-boy** [comment]
@kumarunster could you clarify if you're calling acquireTokenSilent with the scopes in the "https://graph.microsoft.com/openid" "https://graph.microsoft.com/profile" format? Does the loop still happen if you add the scopes as just "openid" and "profile"?

**Comment by kumarunster** [comment]
@technical-boy acquireTokenSilent is invoked inside react-aad-msal, with clientId as the only scope, see [here](https://github.com/syncweek-react-aad/react-aad/blob/5015337ef1eaa6d29822b207ddd2efeedc28caef/packages/react-aad-msal/src/MsalAuthProvider.ts#L160). MSAL 1.4.0 replaces client id with list of default scopes. You can see the scopes in the log file I provided in my last comment:

```
....
VM2349 main.chunk.js:48523 MSAL: - Wed, 26 Aug 2020 09:11:49 GMT:1234-1.4.0-Verbose-pii Serialized scopes: openid profile
VM2349 main.chunk.js:48523 MSAL: - Wed, 26 Aug 2020 09:11:49 GMT:1234-1.4.0-Verbose Account set from MSAL Cache
VM2349 main.chunk.js:48523 MSAL: - Wed, 26 Aug 2020 09:11:49 GMT:1234-1.4.0-Verbose Response type: id_token
VM2349 main.chunk.js:48523 MSAL: - Wed, 26 Aug 2020 09:11:49 GMT:1234-1.4.0-Verbose Finished building server authentication request
VM2349 main.chunk.js:48523 MSAL: - Wed, 26 Aug 2020 09:11:49 GMT:1234-1.4.0-Verbose Query parameters populated from existing SSO or account
VM2349 main.chunk.js:48523 MSAL: - Wed, 26 Aug 2020 09:11:49 GMT:1234-1.4.0-Verbose GetCachedToken has been called
VM2349 main.chunk.js:48523 MSAL: - Wed, 26 Aug 2020 09:11:49 GMT:1234-1.4.0-Verbose Getting all cached access tokens
VM2349 main.chunk.js:48523 MSAL: - Wed, 26 Aug 2020 09:11:49 GMT:1234-1.4.0-Verbose No matching tokens found when filtered by clientId and account
VM2349 main.chunk.js:48523 MSAL: - Wed, 26 Aug 2020 09:11:49 GMT:1234-1.4.0-Verbose No token found in cache lookup
....
```

as explained before, the id token is correctly returned from server, but msal library is not able to find it in local storage.

**Comment by technical-boy** [comment]
Hey, @kumarunster. I can confirm this is in fact a bug in the `acquireTokenSilent` API. I'm working on fixing it. While we get it fixed, do you have a pressing reason to move to `1.4.0` or can you stay at `1.3.x` in the meantime?

**Comment by kumarunster** [comment]
@technical-boy thanks for confirmation. yes we stay with 1.3.4.

**Comment by kumarunster** [comment]
Hello, @technical-boy, @tnorling any issues with PR #2206 ? It is two weeks already since the pr is hanging, is any further help required?

**Comment by technical-boy** [comment]
Hi @kumarunster . We're currently working on testing the changes in the PR, we'll let you know if we need more information. Thanks!

**Comment by hectormmg** [comment]
Closing since msal@1.4.1 has been released with the fix in #2206 . Please let us know if you're still seeing this issue after upgrading to 1.4.1.

**Comment by hectormmg** [comment]
Closing since msal@1.4.1, which includes a fix for this issue, has been published. Please let us know if the issue persists after upgrading to msal@1.4.1.
